devices/Mpu6886/Mpu6886.cs

```diff
@@ -0,0 +1,466 @@
+                v.Y = (short)(vec[2] << 8 | vec[3]);
+                v.Z = (short)(vec[4] << 8 | vec[5]);
+
+                return v;
+            }
+
+            set
+            {
+                SpanByte registerAndOffset = new byte[7];
+                SpanByte offsetbyte = new byte[2];
+
+                registerAndOffset[0] = (byte)Mpu6886.Register.GyroscopeOffsetAdjustmentXHighByte;
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.X);
+                registerAndOffset[1] = offsetbyte[0];
+                registerAndOffset[2] = offsetbyte[1];
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.Y);
+                registerAndOffset[3] = offsetbyte[0];
+                registerAndOffset[4] = offsetbyte[1];
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.Z);
+                registerAndOffset[5] = offsetbyte[0];
+                registerAndOffset[6] = offsetbyte[1];
+
+                _i2c.Write(registerAndOffset);
+            }
+        }
+
+        /// <summary>
+        /// Reset the internal registers and restores the default settings. (Datasheet, page 47)
+        /// </summary>
+        public void Reset()
+        {
+            _i2c.Write(new SpanByte(new byte[] { (byte)Mpu6886.Register.PowerManagement1, 0b1000_0000 }));
+            Thread.Sleep(10);
+        }
+
+        /// <summary>
+        /// Set the chip to sleep mode. (Datasheet, page 47)
+        /// </summary>
+        public void Sleep()
+        {
+            _i2c.Write(new SpanByte(new byte[] { (byte)Mpu6886.Register.PowerManagement1, 0b0100_0000 }));
+            Thread.Sleep(10);
+        }
+
+        /// <summary>
+        /// Disables the sleep mode. (Datasheet, page 47)
+        /// </summary>
+        public void WakeUp()
+        {
+            _i2c.Write(new SpanByte(new byte[] { (byte)Mpu6886.Register.PowerManagement1, 0b0000_0000 }));
+            Thread.Sleep(10);
+        }
+
+        /// <summary>
+        /// Gets and sets the accelerometer full scale. (Datasheet page 37)
+        /// </summary>
+        public AccelerometerScale AccelerometerScale
+        {
+            get
+            {
+                SpanByte buffer = new byte[1];
+                Read(Mpu6886.Register.AccelerometerConfiguration1, buffer);
+                return (AccelerometerScale)(buffer[0] & 0b0001_1000);
+            }
+
+            set
+            {
+                // First read the current register values
+                SpanByte currentRegisterValues = new byte[1];
+                _i2c.WriteByte((byte)Mpu6886.Register.AccelerometerConfiguration1);
+                _i2c.Read(currentRegisterValues);
+
+                byte mask = 0b1110_0111; // we leave all bits except bit 3 and 4 untouched with this mask
+                byte cleaned = (byte)(currentRegisterValues[0] & mask);
+
+                byte newvalue = (byte)(cleaned | (byte)value); // apply the new scale
```

Comment:
To avoid having to allocate 2 bytes which are not need, I would use a condensed version like this:

```suggestion
                byte newvalue = (byte)((currentRegisterValues[0] & 0b1110_0111) | (byte)value); // apply the new scale
```

Please apply this else where. It's important to not allocate anything when not needed.